### ISSUE TITLE
[feature] File selection using emacs

### ISSUE THREAD

**Issue by sarg** [lrustand/qutebrowser.el#4]
File selection could be directed to emacs as well. Here is how I do it:

##  `config.py`
```python
c.fileselect.handler = 'external'
c.fileselect.single_file.command = ['emacsclient', '{}']
c.fileselect.folder.command = ['emacsclient', '{}']
c.fileselect.multiple_files.command = ['emacsclient', '{}']
```

## `init.el`
```lisp
(defun qute/choose-file ()
  (interactive)
  (let ((files (dired-get-marked-files)))
    (with-temp-file qute-filename
      (insert (s-join "\n" files)))
    (unbind-key "C-C C-c" 'dired-mode-map)
    (kill-buffer)))

(defun qute/dired-hook (&optional _)
  (when (s-starts-with? "/tmp/qutebrowser-fileselect" buffer-file-name)
    (setq qute-filename buffer-file-name)
    (kill-buffer)
    (dired "~/")
    (bind-key "C-c C-c" #'qute/choose-file 'dired-mode-map)))

(add-hook! 'server-visit-hook #'qute/dired-hook)
```

**Comment by lrustand** [comment]
Very nice!

I've tried to do something similar previously, but tried to do it globally and not just for Qutebrowser. I don't remember the details, but it turns out that changing the file selector for all applications is non-trivial.

Interesting approach using `server-visit-hook` instead of calling on the functions directly in the emacsclient call. Is there any added benefit behind doing it this way instead of the more direct way?

I don't know what `add-hook!` with an exclamation mark is, but I'm guessing this is something doom specific? Doesn't seem to exist in my emacs at least.

**Comment by sarg** [comment]
Yeah, `add-hook!` is from doom. I think you could simply remove `!` and it should work.

`qutebrowser` waits until the handler exits and expects the selected file name to be written to the provided file. This makes `-e` and lisp magic somewhat troublesome as it would need to block until the user selects the file.

**Comment by lrustand** [comment]
Would you mind making a revised version and sending as a PR?
